lib/puppet/application/lookup.rb

```diff
@@ -49,8 +60,49 @@ class Puppet::Application::Lookup < Puppet::Application
   def run_command
     options[:keys] = command_line.args
 
-    puts "Hello World!"
-    puts options
+    if options[:keys].empty?
+     raise "No keys were given to lookup."
+    end
+
+    if !options[:node]
+      raise "No node was given via the '--node' flag for the scope of the lookup."
+    end
+
+    if (options[:sort_merge_arrays] || options[:merge_hash_arrays] || options[:prefix] || options[:unpack_arrays]) && options[:merge] != 'deep'
+      raise "The options #{DEEP_MERGE_OPTIONS} are only available with '--merge deep'\n#{RUNHELP}"
+    end
+
+    scope = generate_scope
+
+    use_default_value = !options[:default_value].nil?
+    merge_options = nil
+
+    if options[:merge]
+      if options[:merge] == 'deep'
+        merge_options = {'strategy' => 'deep',
+          'sort_merge_arrays' => !options[:sort_merge_arrays].nil?,
+          'merge_hash_arrays' => !options[:merge_hash_arrays].nil?}
+
+        if options[:prefix]
+          merge_options.merge({'prefix' => options[:prefix]})
+        end
+
+        if options[:unpack_arrays]
+          merge_options.merge({'unpack_arrays' => options[:unpack_arrays]})
+        end
+
+      else
+        merge_options = {'strategy' => options[:merge]}
+      end
+    end
+
+    puts Puppet::Pops::Lookup.lookup(scope, options[:keys], options[:type], options[:default_value], use_default_value, {}, {}, merge_options)
   end
 
+  def generate_scope
+    node = Puppet::Node.indirection.find("#{options[:node]}")
+    compiler = Puppet::Parser::Compiler.new(node)
+    compiler.compile
```

Comment:
It will need to do a lot, but not sure it needs to actually evaluate all of the code, produce the catalog etc. While it is possible to create top scope variables when loading code other than the site.pp (or directory representing site.pp), and/or node classifier. I can imagine adding a flag --no-compile that could speed up the lookup by not loading any manifests (i.e. when your data-lookup only depends on the facts).